datadogclient/datadog_client_test.go

```diff
@@ -336,9 +336,40 @@ var _ = Describe("DatadogClient", func() {
 		validateMetrics(payload, 2, 0)
 	})
 
+	It("posts Error Event in JSON format", func() {
+		c.Add(&events.Envelope{
+			Origin:    proto.String("origin"),
+			Timestamp: proto.Int64(1000000000),
+			EventType: events.Envelope_Error.Enum(),
+			Error: &events.Error{
+				Source:  proto.String("app"),
+				Message: proto.String("application failure"),
+				Code:    proto.Int32(500),
+			},
+			Deployment: proto.String("deployment-name"),
+			Job:        proto.String("doppler"),
```

Comment:
This test should demonstrate that envelope tags are being handled correctly; include some on this envelope and assert that they're emitted correctly.